common/djangoapps/third_party_auth/models.py

```diff
@@ -204,6 +212,14 @@ def get_authentication_backend(self):
         """Gets associated Django settings.AUTHENTICATION_BACKEND string."""
         return '{}.{}'.format(self.backend_class.__module__, self.backend_class.__name__)
 
+    @property
+    def display_for_login(self):
+        """
+        Determines whether the provider ought to be shown as an option with
+        which to log in on the login screen, registration screen, and elsewhere.
+        """
+        return bool(self.enabled and self.accepts_logins and self.visible)
```

Comment:
> I guess the real question in my mind is if the resulting cast value will always accurately reflect the combinatorial state we're constructing here.

Yep! Doing a bool cast returns the truthiness value of the contained item - in this case, self.visible if all three items including self.visible are truthy, or the first item that's falsey.